### DISCUSSION TITLE
What do you use for persistent storage?

### DISCUSSION THREAD

**Discussion by nikhilbedi** [MobileNativeFoundation/discussions#160] [General]
There are a variety of persistent storage solutions out there, especially when one is looking to squeeze performance out of reads and/or writes.

What have you found success with? What have you found issues with?

**Comment by nikhilbedi** [comment]
At LinkedIn, we use [LMDB](https://github.com/LMDB/lmdb). Many of our products are read-heavy -- we don't require many writes. LMDB optimizes for the read case, and we definitely see product metrics move when improving read speed of our persistent storage solution.

However, LMDB has proven to be somewhat difficult to work with due to its native access and our relative subpar tooling around analyzing native crashes.

**Reply by nikhilbedi** [reply]
I should mention -- LMDB is what we use in the general case. There are, of course, plenty of cases where we leverage shared prefs/user defaults, and also narrow use cases where we use SQLite wrapped with [FMDB](https://github.com/ccgus/fmdb).

**Comment by DURGESH716** [comment]
Hello @nikhilbedi, Persistent storage is any data storage device that retains data after power to that device is shut off. It is also sometimes referred to as non-volatile storage. Magnetic media, such as hard disk drives and tape are common types of persistent storage, as are the various forms of Optical media such as DVD.

persistent memory (PMEM) is a solid-state high-performance byte-addressable memory device that resides on the memory bus. Being on the memory bus allows PMEM to have DRAM-like access to data, which means that it has nearly the same speed and latency of DRAM and the nonvolatility of NAND flash.
